apps/mark-scan/accessible-controller/src/controllerd.rs

```diff
@@ -178,65 +231,59 @@ fn handle_command(device: &mut Device, data: &[u8]) -> Result<(), ParseError> {
         }
     }
 
-    return send_key(device, key);
+    send_key(device, key)
 }
 
 fn validate_connection(port: &mut Box<dyn SerialPort>) {
-    let echo_command: Vec<u8> = vec![
-        CommandId::Echo as u8,
-        0x00, // u16 data size; fixed value of 5 bytes
-        0x05,
-        0x01, // 5 bytes of arbitrary data
-        0x02,
-        0x03,
-        0x04,
-        0x05,
-        0xfc, // CRC16_XModem value of data packet up to this point
-        0xbd,
-    ];
+    let echo_command = EchoCommand::new(vec![0x01, 0x02, 0x03, 0x04, 0x05]);
+    let echo_command: Vec<u8> = echo_command.into();
     match port.write(&echo_command) {
         Ok(_) => println!("Echo command sent"),
-        Err(error) => eprintln!("{:?}", error),
+        Err(error) => eprintln!("{error:?}"),
     }
 
     let mut serial_buf: Vec<u8> = vec![0; 20];
 
-    // Allow 1s for echo response
-    for _n in 1..(1000 / POLL_INTERVAL_MS) {
+    let start_time = Instant::now();
+    loop {
```

Comment:
I can't try this out so I'm not sure it's right, but this seemed like a better way to poll for up to 1s.